### PR TITLE
Metadata Xml validation

### PR DESCRIPTION
**Pull request by khancyr**
This change the indentation on logger metadata generation to allow validation with xmllint.

The double quote in text has been changed on parameters xml to &quot; . That doesn't impact mavproxy that correctly convert back the value to double quote on printing (there were already other field that were converted). The test was done by replacing the apm.pdef.xml into ~/.mavproxy by the new version and use param help PILOT_THR_BHV for example on Copter.

xmllint check will allow us to check the validity of the xml and thus detect misformed xml that result from error into the parameter/logger descriptions

### CODE REVIEW

**Comment by peterbarker** [comment]
```
Tools/scripts/validate_xml.sh: line 23: xmllint: command not found
```

**Comment by khancyr** [comment]
that is annoying ... then I will try to do the same with lxml ...

**Comment by khancyr** [comment]
@peterbarker any idea to not have to duplicate the code of the validator ? using `from ..param_metadata.xmlvalidator import XmlValidator` doesn't work ....

**Review by peterbarker — DISMISSED** [review]
autotests are broken

**Comment by tridge** [comment]
looks fine to me, but I'll leave it to @peterbarker as this is his area

File: Tools/autotest/logger_metadata/emit_xml.py [hunk 1]
```diff
@@ -3,47 +3,114 @@
...
-            print('        <fields>', file=self.fh)
+            print(INDENT4 + '<fields>', file=self.fh)
             for f in docco.fields_order:
-                print('            <field name="%s">' % f, file=self.fh)
+                print(INDENT6 + '<field name="%s">' % f, file=self.fh)
                 if "description" in docco.fields[f]:
-                    print('                <description>%s</description>' %
-                          docco.fields[f]["description"], file=self.fh)
+                    print(INDENT8 + '<description>%s</description>' %
+                          quoteattr(docco.fields[f]["description"], {'>': "&gt;"}), file=self.fh)
```
Comment (peterbarker) [inline Tools/autotest/logger_metadata/emit_xml.py:43]: I don't understand why we're not always escaping all the same things in quoteattr.  Seems as if we should have our own wrapper around quoteattr which always passes in the extra stuff.

I don't understand why we're passing '>' in here as it appears to be part of the defaults.

Comment (khancyr) [reply Tools/autotest/logger_metadata/emit_xml.py]: I get rid of this madness and make a real xml generation

**Comment by khancyr** [comment]
I have change to use lxml to generate the xml and validate it directly.
The change with master is that non standard characters are now all converted correctly.

This work with Mavproxy and fix some unexpected issues : 
master : 
```
STABILIZE> PILOT_THR_BHV: Throttle stick behavior

Bitmask containing various throttle stick options. TX with sprung throttle can set PILOT_THR_BHV to "1" so motor feedback when landed starts from mid-stick instead of bottom of stick.


Bitmask : 0:Feedback from mid stick,1:High throttle cancels landing,2:Disarm on land detection
```


pr : 
```
STABILIZE> PILOT_THR_BHV: Throttle stick behavior

Bitmask containing various throttle stick options. TX with sprung throttle can set PILOT_THR_BHV to "1" so motor feedback when landed starts from mid-stick instead of bottom of stick.


Bitmask : 0:Feedback from mid stick,1:High throttle cancels landing,2:Disarm on land detection

Values: 
	0 : None
	1 : Feedback from mid stick
	2 : High throttle cancels landing
	4 : Disarm on land detection

```

**Pull request merged**